drivers/serial/uart_mcux_lpuart.c

```diff
@@ -249,13 +252,135 @@ static int mcux_lpuart_init(struct device *dev)
 		return -EINVAL;
 	}
 
+	lpuart_config_t uart_config;
 	LPUART_GetDefaultConfig(&uart_config);
+
+	/* Setup UART init structure */
+
+	/* Translate UART API enum to IMX enum */
+	switch (cfg->parity) {
+	case UART_CFG_PARITY_NONE:
+		uart_config.parityMode = kLPUART_ParityDisabled;
+		break;
+	case UART_CFG_PARITY_ODD:
+		uart_config.parityMode = kLPUART_ParityOdd;
+		break;
+	case UART_CFG_PARITY_EVEN:
+		uart_config.parityMode = kLPUART_ParityEven;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	switch (cfg->data_bits) {
+#if defined(FSL_FEATURE_LPUART_HAS_7BIT_DATA_SUPPORT) && \
+	FSL_FEATURE_LPUART_HAS_7BIT_DATA_SUPPORT
+	case UART_CFG_DATA_BITS_7:
+		uart_config.dataBitsCount  = kLPUART_SevenDataBits;
+		break;
+#endif
+	case UART_CFG_DATA_BITS_8:
+		uart_config.dataBitsCount  = kLPUART_EightDataBits;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+#if defined(FSL_FEATURE_LPUART_HAS_STOP_BIT_CONFIG_SUPPORT) && \
+	FSL_FEATURE_LPUART_HAS_STOP_BIT_CONFIG_SUPPORT
+	switch (cfg->stop_bits) {
+	case UART_CFG_STOP_BITS_1:
+		uart_config.stopBitCount = kLPUART_OneStopBit;
+		break;
+	case UART_CFG_STOP_BITS_2:
+		uart_config.stopBitCount = kLPUART_TwoStopBit;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+#endif
+
+#if defined(FSL_FEATURE_LPUART_HAS_MODEM_SUPPORT) && \
+	FSL_FEATURE_LPUART_HAS_MODEM_SUPPORT
+	switch (cfg->flow_ctrl) {
+	case UART_CFG_FLOW_CTRL_NONE:
+		uart_config.enableTxCTS = false;
+		uart_config.enableRxRTS = false;
+		break;
+	case UART_CFG_FLOW_CTRL_RTS_CTS:
+		uart_config.enableTxCTS = true;
+		uart_config.enableRxRTS = true;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+#endif
+	uart_config.baudRate_Bps = cfg->baudrate;
 	uart_config.enableTx = true;
 	uart_config.enableRx = true;
-	uart_config.baudRate_Bps = config->baud_rate;
+
+	/* disable interrupts */
+	unsigned int old_level = irq_lock();
```

Comment:
Why do you need to disable interrupts?

Reply:
Saw this in the uart_imx driver. Thought that it might be good praxis in case there is anything timing-critical.

Reply:
The lpuart will however be disabled at the time of initialization meaning no uart interrupts should be triggered. This is probably the only thing that would require interrupts to be disabled. For this reason, it might not be desirable to globally disable interrupts for the sake of initializing one lpuart. What do you think?

Reply:
@MaureenHelm Do you think this should be removed?

Reply:
Yes